signer/src/testing/transaction_signer.rs

```diff
@@ -233,6 +235,69 @@ where
         }
     }
 
+    /// Assert that the transaction signer will respond to bitcoin transaction sign requests
+    /// with an acknowledge message
+    pub async fn assert_should_respond_to_bitcoin_transaction_sign_requests(mut self) {
+        let mut rng = rand::rngs::StdRng::seed_from_u64(46);
+        let network = network::in_memory::Network::new();
+
+        let event_loop_harness = EventLoopHarness::create(
+            &mut rng,
+            network.connect(),
+            (self.storage_constructor)(),
+            self.context_window,
+        );
+
+        let mut handle = event_loop_harness.start();
+
+        let test_data = generate_test_data(&mut rng);
+        Self::write_test_data(&test_data, &mut handle.storage).await;
+
+        let coordinator_private_key = p256k1::scalar::Scalar::random(&mut rng);
```

Comment:
Ahh I see. This is all due to the interface in the `p256k1` crate, which is what we sometimes use for generating signatures. To guarantee that changing the seed doesn't error, we'd need to do something like:
```rust
let secret_key = secp256k1::SecretKey::new(&mut rng);
let coordinator_private_key = p256k1::scalar::Scalar::from(secret_key.secret_bytes());
```

Reply:
I'm pretty sure this is infallible. Looking at the documentation of `secp256k1_scalar_set_b32` (which this ends up calling) we have:

So any overflowing numbers will be reduced modulo the group order. Which is actually bad from a statistical perspective because it makes the probability distribution non-uniform, but for testing purposes it's fine.

```
Set a scalar from a big endian byte array. The scalar will be reduced modulo group order `n`.
In:      bin:        pointer to a 32-byte array.
Out:     r:          scalar to be set.
        overflow:   non-zero if the scalar was bigger or equal to `n` before reduction, zero otherwise (can be NULL).
```

Reply:
But thanks for noting this. It could just as well have been a hidden panic.

Reply:
Ahh good call, you are right.